packages/flutter/lib/src/widgets/binding.dart

```diff
@@ -243,6 +243,20 @@ abstract class WidgetsBindingObserver {
   void didChangeAccessibilityFeatures() {}
 }
 
+/// An abstract class for service extension registration.
+///
+/// As BindingBase is a singleton in Flutter framework, and runApp will always ensure there will be
+/// one WidgetsBinding instance. See [WidgetsFlutterBinding.ensureInitialized] for more.
+/// Instances inherited from ServiceExtensionBinding can add itself using [WidgetsBinding.addServiceExtensionBinding]
+/// and will be registered after [WidgetsBinding.initServiceExtensions].
+abstract class ServiceExtensionBinding {
```

Comment:
The name's probably wrong. It's not a 'binding' like the rest of the bindings. It's just a service extension registration (which is not bound yet by the time the user creates it).